### PR TITLE
Plugin Generator Updates

### PR DESCRIPTION
**Pull request by Fryguy**
Plugin Generator Updates

### CODE REVIEW

File: lib/manageiq/schema/engine.rb [hunk 1]
```diff
@@ -3,6 +3,10 @@ module Schema
     class Engine < ::Rails::Engine
       isolate_namespace ManageIQ::Schema
 
+      # This auto-generated line is intentionally commented out to avoid
+      #   autoload of helper classes when part of ManageIQ core.
+      # config.autoload_paths << root.join('lib').to_s
```
Comment (Fryguy) [inline lib/manageiq/schema/engine.rb:8]: I want to get rid of this and use the default, but we have a number of classes in the lib directory that are actually more like helper classes when this plugin is standalone, and so need to be moved / renamed.

File: manageiq-schema.gemspec [hunk 2]
```diff
@@ -1,33 +1,36 @@
...
-  s.add_dependency "pg-pglogical", "~> 2.1.1"
-  s.add_dependency "rails", "~>5.2.4", ">=5.2.4.3"
+  spec.add_dependency "ancestry"
+  spec.add_dependency "activerecord-id_regions", "~> 0.3.0"
+  spec.add_dependency "linux_admin",             "~> 2.0"
+  spec.add_dependency "manageiq-password",       "~> 0.3"
+  spec.add_dependency "more_core_extensions",    ">= 3.5", "< 5"
+  spec.add_dependency "pg"
+  spec.add_dependency "pg-pglogical",            "~> 2.1.1"
+  spec.add_dependency "rails",                   "~>5.2.4", ">=5.2.4.3"
```
Comment (Fryguy) [inline manageiq-schema.gemspec:28]: @jrafanie I think we need to bump this too (and upgrade the dummy app)

**Review by agrare — APPROVED** [review]

**Review by jrafanie — COMMENTED** [review]
LGTM but I can't follow where the changes are coming from?  Are we updating files here that also exist in other repos to keep them in sync here, such as .codeclimate.yml?  Or is there some other reasoning behind the changes?  My assumption and why I'm ok with merging this is I believe it's the former, in that we're just keeping this here up to date with what's changed elsewhere.

**Comment by Fryguy** [comment]
> LGTM but I can't follow where the changes are coming from?

These changes come from the plugin generator in core, and this is the effect of blasting out a re-run of that generator.  This repo is a little different from others though because it doesn't use spec/manageiq, and requires more manual manipulation.

**Comment by jrafanie** [comment]
build error in travis 😢

**Comment by jrafanie** [comment]
```
0.57s$ bin/setup
You are replacing the current local value of path, which is currently nil
CREATE ROLE
Could not find gem 'bundler-inject (~> 1.1)' in any of the gem sources listed in
your Gemfile.
Could not find gem 'rspec' in any of the gem sources listed in your Gemfile.
Run `bundle install` to install missing gems.
The command "bin/setup" failed and exited with 7 during .
```

**Review by jrafanie — APPROVED** [review]

**Comment by agrare** [comment]
All DB related operations are failing after this change, e.g. https://travis-ci.com/github/ManageIQ/manageiq-providers-ibm_cloud/jobs/395145095#L1922

```
== Resetting tests ==
** Confirming rails environment does not connect to the database
rails aborted!
NameError: uninitialized constant ActiveRecord::IdRegions
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/bundler/gems/manageiq-schema-6e9438b3f1e5/config/initializers/activerecord-id_regions.rb:1:in `<top (required)>'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/engine.rb:663:in `block in load_config_initializer'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/activesupport-5.2.4.4/lib/active_support/notifications.rb:170:in `instrument'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/engine.rb:662:in `load_config_initializer'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/engine.rb:620:in `block (2 levels) in <class:Engine>'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/engine.rb:619:in `each'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/engine.rb:619:in `block in <class:Engine>'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:32:in `instance_exec'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:32:in `run'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:61:in `block in run_initializers'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:50:in `each'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:50:in `tsort_each_child'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/initializable.rb:60:in `run_initializers'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/application.rb:361:in `initialize!'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/railtie.rb:190:in `public_send'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/railtie.rb:190:in `method_missing'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/spec/manageiq/config/environment.rb:5:in `<top (required)>'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/application.rb:337:in `require_environment!'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/application.rb:520:in `block in run_tasks_blocks'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/spec/manageiq/lib/tasks/test.rake:23:in `block (3 levels) in <top (required)>'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/spec/manageiq/lib/tasks/evm_rake_helper.rb:7:in `with_dummy_database_url_configuration'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/spec/manageiq/lib/tasks/test.rake:20:in `block (2 levels) in <top (required)>'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/commands/rake/rake_command.rb:23:in `block in perform'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/commands/rake/rake_command.rb:20:in `perform'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/command.rb:48:in `invoke'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/commands.rb:18:in `<top (required)>'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/engine/commands.rb:9:in `require'
/home/travis/build/ManageIQ/manageiq-providers-ibm_cloud/vendor/bundle/ruby/2.5.0/gems/railties-5.2.4.4/lib/rails/engine/commands.rb:9:in `<top (required)>'
bin/rails:14:in `require'
bin/rails:14:in `<main>'
Tasks: TOP => environment => app:environment
(See full trace by running task with --trace)
```

**Pull request merged**